plugin/PDFDetailsPage.qml

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2024 Damien Caliste
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.Office 1.0
+
+DetailsPage {
+    DetailItem {
+        //: Page count of the PDF document
+        //% "Page Count"
+        label: qsTrId("sailfish-office-la-pdf-pagecount")
+        value: document.pageCount
+        alignment: Qt.AlignLeft
+        visible: !document.locked
+    }
+    DetailItem {
+        //% "Read protection"
+        label: qsTrId("sailfish-office-la-pdf-readprotection")
+        value: document.passwordProtected
+            //% "by password"
+            ? qsTrId("sailfish-office-la-pdf-passwordprotected")
+            //% "none"
+            : qsTrId("sailfish-office-la-pdf-noprotection")
+        alignment: Qt.AlignLeft
+        visible: document.passwordProtected
+    }
+    PasswordField {
+        visible: document.passwordProtected
+        enabled: document.password.length > 0
+        x: Theme.horizontalPageMargin
+        width: parent.width - 2*x
+        text: document.password
+        readOnly: true
+    }
+    Button {
+        visible: document.passwordProtected
```

Comment:
This and above could perhaps be hidden when the password is not being remembered.

Reply:
How it is coded now, being remembered is equivalent to password being non empty. So I tried now to bind the opacity to passord.length > 0, with a fade animation. When you look at the details of the document (still locked, or unlocked but not remembered), you just have the read protection notice. If you open a protected document with the cached password, or because you type it and remember it, the password field and the button are shown. When you type on the button, the password is reseted and the password field and the button disappear. Even with the fade animation, that may look a bit strange. Except maybe if I put a remorse, but that starts to be complicated (or not). What do you think ? The remorse may be usefull and would justify the disappearance, like in a item list.

Reply:
Yeh, that's fine like that.